core/corerepo/gc.go

```diff
@@ -14,9 +14,27 @@ type KeyRemoved struct {
 	Key u.Key
 }
 
-func GarbageCollectBlockstore(n *core.IpfsNode, ctx context.Context) (<-chan *KeyRemoved, error) {
+func GarbageCollect(n *core.IpfsNode, ctx context.Context) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel() // in case error occurs during operation
+	keychan, err := n.Blockstore.AllKeysChan(ctx)
+	if err != nil {
+		return err
+	}
+	for k := range keychan { // rely on AllKeysChan to close chan
+		if !n.Pinning.IsPinned(k) {
+			err := n.Blockstore.DeleteBlock(k)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
```

Comment:
im worried that the implementation of `GarbageCollect` and `GarbageCollectAsync` may drift. thoughts?